tests/test_pf7.py

```diff
@@ -0,0 +1,433 @@
+            "NMD": [DIM_VARIANT],
+            "POSITIVE_TRAIN_SITE": [DIM_VARIANT],
+            "QD": [DIM_VARIANT],
+            "QUAL": [DIM_VARIANT],
+            "RAW_MQandDP": [DIM_VARIANT, DIM_PLOIDY],
+            "ReadPosRankSum": [DIM_VARIANT],
+            "RegionType": [DIM_VARIANT],
+            "SOR": [DIM_VARIANT],
+            "VQSLOD": [DIM_VARIANT],
+            "altlen": [DIM_VARIANT, DIM_ALT_ALLELE],
+            "culprit": [DIM_VARIANT],
+            "set": [DIM_VARIANT],
+        }
+
+    @patch("malariagen_data.pf7.init_filesystem", return_value=["fs", "path"])
+    def test_setup_with_config_none(self, mock_init_filesystem):
+        with patch(
+            "malariagen_data.pf7.Pf7._load_config",
+            return_value=self.config_content,
+        ) as mock_load_config:
+            pf7 = Pf7(self.url_starts_with_gs)
+        self.assertEqual(pf7._cache_sample_metadata, None)
+        self.assertEqual(pf7._cache_zarr, None)
+        self.assertEqual(
+            pf7.extended_calldata_variables, self.test_extended_calldata_variables
+        )
+        self.assertEqual(pf7.extended_variant_fields, self.test_extended_variant_fields)
+        mock_load_config.assert_called_once_with(None)
+        mock_init_filesystem.assert_called_once_with(self.url_starts_with_gs)
+
+    @patch("malariagen_data.pf7.init_filesystem", return_value=["fs", "path"])
+    def test_setup_with_config_set(self, mock_init_filesystem):
+        with patch(
+            "malariagen_data.pf7.Pf7._load_config",
+            return_value=self.config_content,
+        ) as mock_load_config:
+            pf7 = Pf7(self.url_starts_with_gs, data_config=self.test_config_path)
+        self.assertEqual(pf7._cache_sample_metadata, None)
+        self.assertEqual(pf7._cache_zarr, None)
+        self.assertEqual(
+            pf7.extended_calldata_variables, self.test_extended_calldata_variables
+        )
+        self.assertEqual(pf7.extended_variant_fields, self.test_extended_variant_fields)
+        mock_load_config.assert_called_once_with(self.test_config_path)
+        mock_init_filesystem.assert_called_once_with(self.url_starts_with_gs)
+
+    def test_load_config(self):
+        config = self.test_pf7_class._load_config(self.test_config_path)
+        self.assertEqual(
+            config,
+            {
+                "metadata_path": "metadata/test_metadata.txt",
+                "zarr_path": "pf7.zarr/",
+            },
+        )
+
+    def test_sample_metadata_returns_extpected_df(self):
+        metadata_df = self.test_pf7_class.sample_metadata()
+        pd.testing.assert_frame_equal(metadata_df, self.test_metadata_df)
+
+    def test_read_general_metadata_returns_expected_df_with_cache_set(
+        self,
+    ):
+        self.test_pf7_class.sample_metadata()
+        metadata_df = self.test_pf7_class.sample_metadata()
+        pd.testing.assert_frame_equal(metadata_df, self.test_metadata_df)
```

Comment:
This tests a public API method and verifies correct behaviour after caching :+1: